src/RedHat.AspNetCore.Server.Kestrel.Transport.Linux/TransportThread.TSocket.cs

```diff
@@ -80,23 +84,40 @@ public struct ReceiveMemoryAllocation
             private int                   _zeropCopyState;
             private SequencePosition      _zeroCopyEnd;
             private int                   _readState = CheckAvailable;
-            public  Task                   MiddlewareTask;
 
             public TSocket(ThreadContext threadContext, int fd, SocketFlags flags)
             {
                 _threadContext = threadContext;
+
+                MaxBufferSize = MemoryPool.MaxBufferSize;
+                BufferMargin = MaxBufferSize / 4;
+
                 Fd = fd;
                 _flags = flags;
                 _onFlushedToApp = new Action(OnFlushedToApp);
                 _onReadFromApp = new Action(OnReadFromApp);
                 _connectionClosedTokenSource = new CancellationTokenSource();
                 ConnectionClosed = _connectionClosedTokenSource.Token;
+                _waitForConnectionClosedTcs = new TaskCompletionSource<object>(TaskCreationOptions.RunContinuationsAsynchronously);
+
                 if (!IsDeferSend)
                 {
                     _sendMemoryHandles = new MemoryHandle[MaxIOVectorSendLength];
                 }
+
+                var inputOptions = new PipeOptions(MemoryPool, PipeScheduler.ThreadPool, PipeScheduler.Inline, PauseInputWriterThreashold, PauseInputWriterThreashold / 2, useSynchronizationContext: false);
+                var outputOptions = new PipeOptions(MemoryPool, PipeScheduler.Inline, PipeScheduler.ThreadPool, PauseOutputWriterThreashold, PauseOutputWriterThreashold / 2, useSynchronizationContext: false);
+
```

Comment:
`PipeScheduler.ThreadPool` should be `PipeScheduler.Inline` when `ApplicationSchedulingMode == PipeScheduler.Inline`